Doc/library/shelve.rst

```diff
@@ -135,6 +149,11 @@ Restrictions
    The *keyencoding* parameter is the encoding used to encode keys before they
    are used with the underlying dict.
 
+   The *deserializer* parameter can be the function that takes the
+   :term:`bytes-like object` and the *protocol* parameter and returns the
+   object. *serializer* parameter can be the function that takes the object
+   and returns :class:`bytes`.
+
```

Comment:
These only need one description:
```suggestion
   The *deserializer* and *serializer* are as in :func:`~shelve.open`.

```
(If you accept this, also replace `Shelve` by `shelve.open` in `BsdDbShelf` & `DbfilenameShelf` below.)